pipeline_dp/dp_engine.py

```diff
@@ -110,3 +114,20 @@ def unnest_cross_partition_bound_sampled_per_key(pid_pk_v):
         return self._ops.flat_map(col,
                                   unnest_cross_partition_bound_sampled_per_key,
                                   "Unnest")
+    
+    def _select_private_partitions(self, col):
+        """
+        Selects and publishes private partitions.
+
+        Args:
+            col: collection, with types for each element: 
+                (partition_key, privacy_id_count, value)
+        
+        Returns:
+            collection of elements (partition_key, data)
+        """
+        def filter_fn(row):
+            return self._partition_selection_strategy.should_keep(row[1])
```

Comment:
As mentioned earlier we need to create PartitionSelectionStrategy in this function, there are some nuances.

1.For now let's use only `TruncatedGeometric` selection strategy (budget accounting is not ready yet for other methods).
2.`TruncatedGeometric` requires (eps, delta, max_partitions_contributed)
  - please add `max_partitions_contributed` to arguments of  `_select_private_partitions`
  - (eps, delta) are computed lazily after the pipeline is constructed (and before it started executing on workers), which means that PartitionSelectionStrategy should be created lazily, namely
      1. By calling self._budget_accountant.[request_budget()](https://github.com/OpenMined/PipelineDP/blob/5f8e2b5e1265af6e29f3c2fb79d0c9846a63ba43/pipeline_dp/budget_accounting.py#L72), you can get `Budget` object, but budget.eps, budget.delta will be available only during the running (e.g. when `filtern_fn` is called)
      2. In `filter_fn` it's possible to create `PartitionSelectionStrategy` and use it (I'm not sure how expensive is creating PyBind11 object, we can start from creating it on each call of `filter_fn` and if it's expensive to figure out how to create it only once per a worker)